src/org/intellij/erlang/compilation/ErlangPrepareDependenciesCompileTask.java

```diff
@@ -140,55 +148,95 @@ private static ErlangModuleBuildOrderDescriptor getModuleBuildOrderInner(Module
-                                       @NotNull Collection<VirtualFile> erlangHeaders,
-                                       @NotNull List<String> globalParseTransforms,
-                                       @NotNull PsiManager psiManager) {
-      myPsiManager = psiManager;
-      buildNamesMap(erlangModules);
-      buildForHeaders(erlangHeaders);
-      buildForModules(erlangModules, globalParseTransforms);
+    @Override
+    public Collection<String> getNodes() {
+      return myPathsToDependenciesMap.keySet();
+    }
+
+    @Override
+    public Iterator<String> getIn(@NotNull String filePath) {
+      return myPathsToDependenciesMap.get(filePath).iterator();
     }
 
     @NotNull
-    public static ErlangFilesDependencyGraph createSemiGraph(@NotNull Module module,
-                                                             boolean onlyTestModules) {
-      Collection<VirtualFile> erlangModules = ErlangModulesUtil.getErlangModuleFiles(module, onlyTestModules);
-      Collection<VirtualFile> erlangHeaders = ErlangModulesUtil.getErlangHeaderFiles(module, onlyTestModules);
+    public List<String> getDependencies(@NotNull String filePath) {
+      List<String> dependencies = myPathsToDependenciesMap.get(filePath);
+      assert dependencies != null;
+      return dependencies;
+    }
 
-      return new ErlangFilesDependencyGraph(erlangModules, erlangHeaders, getGlobalParseTransform(module), PsiManager.getInstance(module.getProject()));
+    private void buildDependenciesMap(@NotNull Module module) {
+      List<String> globalParseTransform = getGlobalParseTransform(module);
+      buildDependenciesMap(module, ErlangModulesUtil.getErlangHeaderFiles(module, false), ContainerUtil.<String>emptyList());
+      buildDependenciesMap(module, ErlangModulesUtil.getErlangHeaderFiles(module, true), ContainerUtil.<String>emptyList());
+      buildDependenciesMap(module, ErlangModulesUtil.getErlangModuleFiles(module, false), globalParseTransform);
+      buildDependenciesMap(module, ErlangModulesUtil.getErlangModuleFiles(module, true), globalParseTransform);
     }
 
-    private void buildForModules(@NotNull Collection<VirtualFile> erlangModules,
-                                 @NotNull List<String> globalParseTransforms) {
+    private void buildDependenciesMap(@NotNull Module module, @NotNull Collection<VirtualFile> erlangModules,
+                                      @NotNull List<String> globalParseTransforms) {
       for (VirtualFile erlangModule : erlangModules) {
         Set<String> dependencies = ContainerUtil.newHashSet();
         ErlangFile erlangFile = getErlangFile(erlangModule);
-        addDeclaredDependencies(erlangFile, dependencies);
-        dependencies.addAll(getPathsFromNames(globalParseTransforms));
+        addDeclaredDependencies(module, erlangFile, dependencies);
+        dependencies.addAll(resolvePathsFromNames(globalParseTransforms, module));
```

Comment:
There is no need to resolve these paths on every iteration - you only need it resolved once. Moreover, you can resolve it once one level above, passing resolved parse transforms to this method.